docs/work-avoidance.md

```diff
@@ -2,7 +2,11 @@
 
 > v2.9 and after
 
-You can make workflows faster and more robust by employing **work avoidance**. A workflow that utilizes this is simply a workflow containing steps that do not run if the work has already been done. This simplest way to do this is to use **marker files**.
+You can make workflows faster and more robust by employing **work avoidance**. A workflow that utilizes this is simply a workflow containing steps that do not run if the work has already been done.
+
+This technique is similar to [memoization](memoization.md) but they have distinct use cases. Work avoidance is totally in your control and you make the decisions as to have to skip the work. [Memoization](memoization.md) is a feature of Argo Workflows to automatically skip steps which generate outputs - it is designed
```

Comment:
The last sentence here was incomplete; it ends with "it is designed". Not sure how it was intended to be written

Reply:
Oh, erm... yeah, I'll finish that off. Thanks @agilgur5

Reply:
Fixed in #11357 . Thanks!